Refactor codes relevant to pretrain

seopbo opened the pull request:
- 학습하고자하는 plm 별로 `DataCollatorFor{MODEL}`을 추가함.
- pretrain_language_model.py에서 `model_type_to_collator`를 정의하여,
  `model_type` 별로 collator를 가져옴.
  - config 파일의 `collator` 항목에서 collator를 위한 args (e.g. `mlm_probability`)를 가져옴.
- pretrain_language_model.py에서 eval_dataset을 사용하기위한 코드추가
  - config 파일의 `data` 항목에서 eval_dataset을 설정하기위한 `test_size` arg를 가져옴.
- 그 의 isort, black 돌림.


Refs: #30

bzantium reviewed (approved):
LGTM 깔끔하네요!

monologg commented:
@bzantium 
`null` 이랑 `False` 가 다르게 인식되는건 처음 알았네요 (하이드라가 아직 익숙치가 않아서...)
시간나는대로 다른 부분도 체크해볼게요 👍

seopbo committed:
[python] refactor: Remove electra
- 이번 release에서 전체적으로 지원하지않아서 삭제함.

Refs: #30

seopbo committed:
[mixed] Refactor: Add DataCollator classes
- 학습하고자하는 plm 별로 DataCollatorFor{MODEL}을 추가함.
- pretrain_language_model.py에서 model_type_to_collator를 정의하여,
  model_type 별로 collator를 가져옴.
- config 파일의 collator 항목에서 collator를 위한 args를 가져옴.
- 그 의 isort, black 돌림.

Refs: #30

bzantium commented:
만약을 대비해서 Collator __init__에 **kwargs을 추가하는 것은 어떨까요?

seopbo committed:
[python] fix: Fix setting args codes
- data_args를 그대로 이용하게끔 수정함.

Refs: #30

seopbo commented:
> 만약을 대비해서 Collator __init__에 **kwargs을 추가하는 것은 어떨까요?

그 방식으로 처음에 진행을 했었는데요. 예컨대 상속받아서 클래스를 선언해두고 부모의 `__init__`을 사용하는 경우, 어차피 `DataCollatorForLanguageModeling`이 `**kwargs`를 받지않아서 사용이 어려웠습니다.

to: @bzantium 
cc: @monologg @iron-ij @alxiom

seopbo commented:
> 만약을 대비해서 Collator __init__에 **kwargs을 추가하는 것은 어떨까요?

개인적으로 config 파일에서 `collator` 항목을 분기하면서. `collator` 항목 하위에 `model_type`을 두고, `model` 항목 하위의 `model_type`가 `==` check를 하는 것이 좀 고민되는데, 이것에 대해서는 어떻게 생각하시나요?

cc: @bzantium @alxiom @monologg @iron-ij

bzantium commented:
> 그 방식으로 처음에 진행을 했었는데요. 예컨대 상속받아서 클래스를 선언해두고 부모의 `__init__`을 사용하는 경우, 어차피 `DataCollatorForLanguageModeling`이 `**kwargs`를 받지않아서 사용이 어려웠습니다.
> 
> to: @bzantium cc: @monologg @iron-ij @alxiom

`super().__init__()` 에 `kwargs`를 주입안하면 문제없지 않을까요? 사실 config에서 딱 맞춰서 주면 문제가 없긴합니다 :) 또한
train_tokenizer.py 에서 `{"bert": "bert-base-cased"}`로 규정하는 것은 어떨까요? 어제 이야기한 것 처럼 uncased는 제외하는 것.

to: @seopbo 
cc: @alxiom @monologg @iron-ij

bzantium commented:
> 개인적으로 config 파일에서 `collator` 항목을 분기하면서. `collator` 항목 하위에 `model_type`을 두고, `model` 항목 하위의 `model_type`가 `==` check를 하는 것이 좀 고민되는데, 이것에 대해서는 어떻게 생각하시나요?
> 
> cc: @bzantium @alxiom @monologg @iron-ij

저는 개인적으로는 지금 코드를 유지하는 것이 좋을 것 같다는 생각입니다!

monologg commented:
> > 그 방식으로 처음에 진행을 했었는데요. 예컨대 상속받아서 클래스를 선언해두고 부모의 `__init__`을 사용하는 경우, 어차피 `DataCollatorForLanguageModeling`이 `**kwargs`를 받지않아서 사용이 어려웠습니다.
> > to: @bzantium cc: @monologg @iron-ij @alxiom
> 
> `super().__init__()` 에 `kwargs`를 주입안하면 문제없지 않을까요? 사실 config에서 딱 맞춰서 주면 문제가 없긴합니다 :) 또한 train_tokenizer.py 에서 `{"bert": "bert-base-cased"}`로 규정하는 것은 어떨까요? 어제 이야기한 것 처럼 uncased는 제외하는 것.
> 
> to: @seopbo cc: @alxiom @monologg @iron-ij

bert는 `bert-base-cased`만 우선 지원하게 하면 좋을 것 같습니다:)

seopbo committed:
Update pretrain_language_model.py

Co-authored-by: Jangwon Park <adieujw@gmail.com>

bzantium reviewed (commented):
suggestion 코드와 같은 방식으로 요구하는 키워드 외에 다른 argument가 들어와도 오류없이 처리할 수 있도록 하자는 의미입니다!

seopbo committed:
[python] fix: Fix pandas ver
- poetry install 시, pandas ver에 의한 bug를 fix

Refs: #30

seopbo committed:
[python] refactor: Refactor codes along comments
- 장원님이 리뷰해주신 내용 반영함.

Refs: #30

seopbo committed:
[python] style: Applying formatting
- black, isort가 적용되지않은 스크립트에 대하여 적용함.

Refs: #30

seopbo committed:
[python] feat: Add a pipreqs package
- poetry를 사용하지않을 경우, requirements.txt를 뽑아주기위한 package를
  추가함.

Refs: #30

seopbo committed:
[python] chore: Add requirements.txt
- poetry를 사용하지않을 경우 의존성에 대한 정보 추가

Refs: #30

bzantium reviewed (approved):
LGTM :)

iron-ij reviewed (approved):
LGTM 👍
